android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/UrlDeviceDiscoveryService.java

```diff
@@ -241,13 +240,12 @@ private void cancelNotifications() {
 
   private void saveCache() {
     // Write the PW Collection
-    String preferencesKey = getString(R.string.discovery_service_prefs_key);
-    SharedPreferences prefs = getSharedPreferences(preferencesKey, Context.MODE_PRIVATE);
-    SharedPreferences.Editor editor = prefs.edit();
-    editor.putInt(PREFS_VERSION_KEY, PREFS_VERSION);
-    editor.putLong(SCAN_START_TIME_KEY, mScanStartTime);
-    editor.putString(PW_COLLECTION_KEY, mPwCollection.jsonSerialize().toString());
-    editor.apply();
+    SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(this);
```

Comment:
just put this on one line: PreferenceManager.getDefaultSharedPreferences(this).edit() (etc)